### PR TITLE
Fix bug 1315082: Reset git repos on failed db updates

### PR DESCRIPTION
**Pull request by pmac**
Reset git repos from git-based data import commands (e.g. update_security_advisories) when data import fails but the git update succeeds. In this way the next run of the command will try again and see all of the changed files.

### CODE REVIEW

**Comment by pmac** [comment]
@jgmize this is a start on improving the git-based commands. I need to test, but figured I'd get this in so we could discuss direction.

**Comment by jgmize** [comment]
Seems like a plausible approach to me. Looking forward to seeing your testing approach for the failure scenario this is supposed to address.

**Comment by pmac** [comment]
You and me both

File: bedrock/mozorg/management/commands/update_product_details_files.py [hunk 1]
```diff
@@ -53,13 +54,20 @@ def handle(self, *args, **options):
...
+        if options['force'] or modified is None:
             files_to_load = self.file_storage.all_json_files()
         else:
             files_to_load = self.filter_filenames(modified)
 
         if files_to_load:
-            self.load_changes(options, files_to_load)
+            try:
+                self.load_changes(options, files_to_load)
+            except Exception:
```
Comment (metadave) [inline bedrock/mozorg/management/commands/update_product_details_files.py]: is it worth logging the exception here?

Comment (pmac) [inline bedrock/mozorg/management/commands/update_product_details_files.py]: Could do. I'll add that. Thanks!

**Comment by pmac** [comment]
@jgmize I've updated this with the new DB-based directly. It may well be ready to go. This also decently simplifies the product-details and security-advisory commands.

File: bedrock/utils/migrations/0001_initial.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+    ]
+
+    operations = [
+        migrations.CreateModel(
+            name='GitRepoState',
```
Comment (pmac) [inline bedrock/utils/migrations/0001_initial.py]: I have added a call to `manage.py migrate` to the clock process startup.

**Comment by pmac** [comment]
@jgmize ready for another (and possibly final) look.

**Comment by pmac** [comment]
@jgmize I changed the call in supervisor to `run-clock.sh`. Should be good now I think.

**Pull request merged**